src/libraries/System.Net.Sockets/tests/FunctionalTests/Accept.cs

```diff
@@ -330,20 +340,33 @@ public async Task AcceptGetsCanceledByDispose()
                     disposedException = true;
                 }
 
-                if (UsesApm)
-                {
-                    Assert.Null(localSocketError);
-                    Assert.True(disposedException);
-                }
-                else if (UsesSync)
+                try
                 {
-                    Assert.Equal(SocketError.Interrupted, localSocketError);
+                    if (UsesApm)
+                    {
+                        Assert.Null(localSocketError);
+                        Assert.True(disposedException);
+                    }
+                    else if (UsesSync)
+                    {
+                        Assert.Equal(SocketError.Interrupted, localSocketError);
+                    }
+                    else
+                    {
+                        Assert.Equal(SocketError.OperationAborted, localSocketError);
+                    }
+                    break;
                 }
-                else
+                catch
```

Comment:
Should we catch a specific exception here, like some XUnit exception class? Seems like that would be safer and clearer. Similarly below.